lib/js/app/queryCreator/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { FC, useEffect, useRef, useMemo, useReducer } from 'react';
+import { useSelector } from 'react-redux';
+import shallowEqual from 'shallowequal';
+import { Button, Select, Input } from '@keen.io/ui-core';
+import { DatePicker } from './DatePicker';
+import { GeoCoordinates } from './GeoCoordinates';
+
+import {
+  addFilter,
+  removeFilter,
+  updateFilter,
+  resetFilters,
+  setFilters
+} from './actions';
+import { getCollectionSchema } from '../../modules/events';
+
+import { filtersReducer } from './reducer';
+
+import { DATA_TYPES, FILTER_OPERATORS, DEFAULT_TIMEFRAME_ABSOLUTE_VALUE } from './constants';
+import { convertFilters, getTypeFromValue, convertSchemaProp } from './utils';
+
+import { AppState, PropertyType, Operator, Filter } from '../../types';
+import { SchemaProp } from './types';
+
+import text from './text.json';
+
+type Props = {
+  /** Collection name */
+  collection: string;
+  /** Filters */
+  filters: Filter[];
+  /** Onchange handler */
+  onChange: (filters: Filter[]) => void;
+};
+
+const Filters: FC<Props> = ({ collection, filters }) => {
+  const collectionSchema = useSelector((state: AppState) =>
+    getCollectionSchema(state, collection)
+  );
+
+  const options = useMemo(() => {
+    if (collectionSchema) {
+      return Object.keys(collectionSchema).map((propertyName) => ({
+        label: propertyName,
+        value: propertyName,
+      }));
+    }
+
+    return [];
+  }, [collectionSchema]);
+
+  const dataTypes = Object.keys(DATA_TYPES).map(item => ({ label: DATA_TYPES[item], value: DATA_TYPES[item] }));
+
+  const getOperatorOptions = (type?: PropertyType) => {
+    if (!type) return FILTER_OPERATORS;
+
+    return FILTER_OPERATORS.filter(operator => operator.dataTypes.includes(type));
+  }
+
+  const [state, filtersDispatcher] = useReducer(filtersReducer, []);
+  const stateRef = useRef(state);
+
+  const filtersRef = useRef(filters);
+
+  useEffect(() => {
```

Comment:
I'm not sure if we need to perform `shallowEqual` for filters. As they cannot be changed in time by selecting another query. 

So during the component mount we could just set local reducer state.